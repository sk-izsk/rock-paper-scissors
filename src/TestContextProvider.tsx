import React from 'react';
import { ThemeProvider } from 'react-jss';
import { theme } from './theme';

const TestContextProvider: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export { TestContextProvider };
