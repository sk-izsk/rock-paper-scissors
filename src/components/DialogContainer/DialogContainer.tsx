import clsx from 'clsx';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { Dialog } from 'ui-neumorphism';
import { CustomTheme } from '../../theme';
import { CardContainer } from '../CardContainer/CardContainer';

export interface DialogContainerProps {
  visible?: boolean;
  onClose?: () => void;
  dialogStyle?: string;
  inset?: boolean;
  cardContentStyle?: string;
  cardAction?: JSX.Element;
  cardStyle?: string;
  cardLoading?: boolean;
  cardBordered?: boolean;
}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  modalContainer: {
    maxWidth: 500,
    width: '100%',
    padding: theme.spacing(1),
  },
}));

const DialogContainer: React.FC<DialogContainerProps> = ({
  visible,
  onClose,
  children,
  dialogStyle,
  inset,
  cardContentStyle,
  cardAction,
  cardLoading,
  cardStyle,
  cardBordered,
}) => {
  const classes = useStyles();
  return (
    <Dialog
      className={clsx([classes.modalContainer, dialogStyle && dialogStyle])}
      visible={visible}
      onClose={onClose}
      persistent={true}
    >
      <CardContainer
        cardStyle={cardStyle}
        cardLoading={cardLoading}
        cardContentStyle={cardContentStyle}
        inset={inset}
        cardAction={cardAction}
        cardBordered={cardBordered}
      >
        {children}
      </CardContainer>
    </Dialog>
  );
};

export { DialogContainer };
