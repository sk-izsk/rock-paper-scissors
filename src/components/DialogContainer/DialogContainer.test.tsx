import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { DialogContainer } from './DialogContainer';

const mockData = {
  inset: true,
  cardAction: <div>mock</div>,
  cardLoading: true,
  cardBordered: true,
  visible: true,
  onClose: jest.fn(),
};

test('DialogContainer renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <DialogContainer {...mockData} />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
