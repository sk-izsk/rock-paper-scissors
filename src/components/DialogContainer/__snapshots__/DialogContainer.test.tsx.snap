// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogContainer renders correctly 1`] = `<DocumentFragment />`;
